cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaMirrorMaker2AssemblyOperator.java

```diff
@@ -0,0 +1,374 @@
+                jaasConfig = "org.apache.kafka.common.security.scram.ScramLoginModule required username=\"" + authProperties.get(AuthenticationUtils.SASL_USERNAME) + "\" password=\"${file:" + CONNECTORS_CONFIG_FILE + ":" + cluster.getAlias() + ".sasl.password}\";";
+            } else if (KafkaClientAuthenticationOAuth.TYPE_OAUTH.equals(clientAuthType)) {
+                KafkaClientAuthenticationOAuth oauth  = (KafkaClientAuthenticationOAuth) cluster.getAuthentication();
+
+                StringBuilder oauthJaasConfig = new StringBuilder();
+                oauthJaasConfig.append("org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required ");
+
+                if (authProperties.containsKey("OAUTH_CONFIG")) {
+                    oauthJaasConfig.append(authProperties.get("OAUTH_CONFIG"));
+                }
+
+                if (oauth.getClientSecret() != null) {
+                    oauthJaasConfig.append(" oauth.client.secret=\"${file:" + CONNECTORS_CONFIG_FILE + ":" + cluster.getAlias() + ".oauth.client.secret}\"");
+                }
+
+                if (oauth.getAccessToken() != null) {
+                    oauthJaasConfig.append(" oauth.access.token=\"${file:" + CONNECTORS_CONFIG_FILE + ":" + cluster.getAlias() + ".oauth.access.token}\"");
+                }
+
+                if (oauth.getRefreshToken() != null) {
+                    oauthJaasConfig.append(" oauth.refresh.token=\"${file:" + CONNECTORS_CONFIG_FILE + ":" + cluster.getAlias() + ".oauth.refresh.token}\"");
+                }
+
+                if (oauth.getTlsTrustedCertificates() != null && !oauth.getTlsTrustedCertificates().isEmpty()) {
+                    oauthJaasConfig.append(" oauth.ssl.truststore.location=\"/tmp/kafka/clusters/" + cluster.getAlias() + "-oauth.truststore.p12\" oauth.ssl.truststore.password=\"${file:" + CONNECTORS_CONFIG_FILE + ":oauth.ssl.truststore.password}\" oauth.ssl.truststore.type=\"PKCS12\"");
+                }
+
+                oauthJaasConfig.append(";");
+
+                saslMechanism = "OAUTHBEARER";
+                jaasConfig = oauthJaasConfig.toString();
+                config.put(configPrefix + SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS, "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler");
+            }
+
+            if (saslMechanism != null) {
+                config.put(configPrefix + SaslConfigs.SASL_MECHANISM, saslMechanism);
+            }
+            if (jaasConfig != null) {
+                config.put(configPrefix + SaslConfigs.SASL_JAAS_CONFIG, jaasConfig);
+            }
+        }
+
+        if (securityProtocol != null) {
+            config.put(configPrefix + AdminClientConfig.SECURITY_PROTOCOL_CONFIG, securityProtocol);
+        }
+
+        config.putAll(cluster.getConfig().entrySet().stream()
+                .collect(Collectors.toMap(entry -> configPrefix + entry.getKey(), Map.Entry::getValue)));
+        config.putAll(cluster.getAdditionalProperties());
```

Comment:
Reading this I can't help but feel we need to factor out some class for writing JAAS configs. I think a recent PR from @scholzj also did some JAAS config generation. Having a dedicated class would mean we could have specific tests for it. @ajborley you don't have to do this in this PR, but if we agree then could you open an issue for this to be done in the future.

Reply:
There is some JAAS config generation in `KafkaBrokerConfigurationBuilder.java` although most of it is in tests and the *_run.sh scripts. I will open an issue.